### PR TITLE
Feature/add basic UI

### PR DESCRIPTION
**Pull request by techEmprez**
### 
I implemented...
- Created an app.rb file that serves as the console app entry-point. It has methods that do the following:
- List all books.
- List all people.
- Create a person (teacher and student)
- Create a book.
- Create a rental.
- List all rentals for a given person id.
- Created a main.rb file that defines the entry point
- Present the user with a list of options to perform.
- Lets users choose an option.
- Ask for parameters for the option.
- Quits the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    when 2
+      list_all_people
+    when 3
+      create_a_person
+    when 4
+      create_a_book
+    when 5
+      create_a_rental
+    when 6
+      list_all_rentals
```
Comment (HAMMAS-SALEEM) [inline app.rb]: ### Error on Option 1
image

### Error on Option 3
image

### Error on Option 4
image

### Error on Option 5
image

- [ ] As you can see in the above images that on some options of your UI we get errors. Kindly fix those errors to make your application work as expected :blush:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @techEmprez :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Github Flow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      @people.push(Student.new('classroom', name, age))
+      puts "#{name.capitalize} was added as a student successfully"
+    when 2
+      print 'Enter the teacher name: '
+      name = gets.chomp
+      print 'Enter the teacher\'s age: '
+      age = gets.chomp
+      print 'Enter the teacher\'s specialization: '
+      specialization = gets.chomp
+      @people.push(Teacher.new(specialization, name, age))
```
Comment (rloterh) [inline app.rb:48]: - [ ] Your app throws an error just after entering a teacher's details, under option 3, after implementing changes. Kindly ensure to improve upon the same.

**Error entering teacher's details:**
image

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    puts 'Person selected'
+    print 'Date: '
+    date = gets.chomp
+    puts 'Date of renting book is added'
+    @rentals.push(Rental.new(date, @people[person_number - 1], @books[book_number - 1]))
+    puts 'Rental book created successfully'
+  end
+
+  # LIST ALL RENTALS FOR A GIVEN PERSON ID
+  def list_all_rentals
```
Comment (rloterh) [inline app.rb:87]: - [ ] Again, attempt to list all rentals for a given person id is throwing an error, after implementing changes. Kindly ensure to improve upon the same.
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Choice, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job with improvements so far

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly include instructions in your ReadMe on how to setup and run your project, under the _Getting Started_ section.
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor. 
- Issues link in your ReadMe is broken, kindly endeavor to fix it.
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @techEmprez, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] Kindly fix the changes asked by previous reviewers. [here](https://github.com/techEmprez/OOP_SCHOOL_LIBRARY/pull/4/files#r1014674960)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @techEmprez, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**